pkg/services/ngalert/store/provisioning_store.go

```diff
@@ -76,3 +93,15 @@ func (st DBstore) SetProvenance(ctx context.Context, o models.Provisionable, p m
 		return nil
 	})
 }
+
+// DeleteProvenance deletes the provenance record from the table
+func (st DBstore) DeleteProvenance(ctx context.Context, orgID int64, o models.ProvisionableInOrg) error {
```

Comment:
The `SetProvenance` API deletes the provenance, and then re-creates it under the new value. I wonder if instead of needing this, we could just wrap the insert in `SetProvenance` with an `if p != models.ProvenanceNone`? Then, you could just delete the provenance for any object by setting it to ProvenanceNone. IMO that's a bit more consistent with how this component's API works, because it's assumed to be None already if there is no record for it - so the concept of Delete doesn't make as much sense on its own.

Reply:
We talked about this offline, we won't change it for now as it might not be obvious how to delete a provenance.